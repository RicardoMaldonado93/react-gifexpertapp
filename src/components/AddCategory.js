import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategory}) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target}) =>setInputValue(target.value);

    const handleSubmit = (e)=>{ 
        e.preventDefault(); 

        if(inputValue.trim().length > 2){
            setCategory( items =>[  inputValue, ...items ]);
            setInputValue("");
            // document.querySelector("input").placeholder = "ingrese categoria..."
        }

    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}

