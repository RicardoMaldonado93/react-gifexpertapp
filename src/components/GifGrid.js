import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    const { data:images, loading } = useFetchGif( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> ~ { category } ~ </h3>
            { loading && <p className="animate__animated animate__flash">cargando...</p> }
            <div className="card-grid animate__animated animate__fadeIn">
                {         
                    images.map( img =>
                    <GifGridItem 
                        {...img}
                        key={img.id}
                    />  
                    )
                }
            </div>
        </>
    )
}
