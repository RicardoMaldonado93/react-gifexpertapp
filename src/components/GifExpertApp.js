import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] })=>{
    
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategory={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category =>
                        <GifGrid 
                            category={category}
                            key={category}
                        />
                    )
                }

            </ol>
            
        </>
    );
}

export default GifExpertApp;