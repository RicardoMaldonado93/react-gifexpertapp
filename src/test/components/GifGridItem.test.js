import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("unit test for '<GifGridItem />'", () => {
    const title = "title";
    const url = "https://localhost/any.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test("should render component correctly ", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should contain a tag <p>",() =>{
      const p = wrapper.find("p").exists();
      expect( p ).toBeTruthy();
    });

    test("should contain a tag <img>",() =>{
      const img = wrapper.find("img").exists();
      expect( img ).toBeTruthy();
    });

    test("tag '<p>' should contain a title",() =>{
      const p = wrapper.find('p');
      expect( p.text().trim() ).toBe( title );
    });

    test("tag '<img>' should contain a src resource",() =>{
      const img = wrapper.find('img').prop("src");
      expect( img.trim() ).toBe(url);
    });

    test("tag '<img>' should contain an alt resource equals to title prop",() =>{
      const img = wrapper.find('img').prop("alt")
      expect( img.trim() ).toBe(title);
    });

    test("'<div>' should contain the className 'animate__fadeIn'",() =>{
      const div = wrapper.find('div')
      const className = div.prop("className");

      expect( div.hasClass("animate__fadeIn") ).toBeTruthy();
      expect( className.includes("animate__fadeIn") ).toBeTruthy();
    });
})
