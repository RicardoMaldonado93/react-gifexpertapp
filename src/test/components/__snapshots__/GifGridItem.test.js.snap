// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test for '<GifGridItem />' should render component correctly  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost/any.jpg"
  />
  <p>
    title
  </p>
</div>
`;
